packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -159,15 +167,73 @@ export const useShippingMethod = props => {
         // Determine the component's display state.
         const nextDisplayState = selectedMethod
             ? displayStates.DONE
+            : isBackgroundAutoSelecting
+            ? displayStates.INITIALIZING
             : displayStates.EDITING;
         setDisplayState(nextDisplayState);
-    }, [data]);
+    }, [data, isBackgroundAutoSelecting]);
+
+    // If an authenticated user does not have a preferred shipping method,
+    // auto-select the least expensive one for them.
+    useEffect(() => {
```

Comment:
This is the meat of the PR, everything else is minor changes to support a seamless view for the user.